inc/namespace.php

```diff
@@ -548,3 +561,65 @@ function add_ec2_instance_data_to_xray( array $trace ) : array {
 
 	return $trace;
 }
+
+/**
+ * Return an AWS CloudFront Client instance
+ *
+ * @return \Aws\CloudFront\CloudFrontClient
+ */
+function get_cloudfront_client() : CloudFrontClient {
+	return get_aws_sdk()->createCloudFront( [
+		'version' => '2019-03-26',
+	] );
+}
+
+/**
+ * Create purge request to invalidate CDN cache.
+ *
+ * @param array $paths_patterns a list of the paths that you want to invalidate.
+ *                              The path is relative to the CDN host, A leading / is optional.
+ *                              e.g  for http://altis-dxp.com/images/image2.jpg
+ *                              specify images/image2.jpg or /images/image2.jpg
+ *
+ *                              You can also invalidate multiple files simultaneously by using the * wildcard.
+ *                              The *, which replaces 0 or more characters, must be the last character in the invalidation path.
+ *                              e.g /images/* - will invalidate all files in a directory
+ *
+ * @return bool return true if invalidation successfully created, false on failure.
+ */
+function purge_cdn_paths( array $paths_patterns ): bool {
+	$client = get_cloudfront_client();
+
+	$distribution_id = '';
+
+	if ( defined( 'CLOUDFRONT_DISTRIBUTION_ID' ) ) {
+		$distribution_id = CLOUDFRONT_DISTRIBUTION_ID;
+	}
+
+	$distribution_id = apply_filters( 'altis.cloud.cdn_distribution_id', $distribution_id );
+
+	if ( empty( $distribution_id ) ) {
+		trigger_error( 'Empty cloudfront distribution id for purge request.', E_USER_WARNING );
+
+		return false;
+	}
+
+	try {
+		$client->createInvalidation( [
+			'DistributionId'    => $distribution_id,
+			'InvalidationBatch' => [
+				'Paths'           => [
+					'Items'    => $paths_patterns,
+					'Quantity' => count( $paths_patterns ),
+				],
+				'CallerReference' => md5( time() . wp_json_encode( $paths_patterns ) ),
```

Comment:
```suggestion
				'CallerReference' => sha1( time() . wp_json_encode( $paths_patterns ) ),
```
We've been moving to using `sha1()` for hashing instead as it's less prone to collisions.